File: releases/Quicksplash-v1.2/Samples/v2.0/VB-ProgressBarAndStatusLabel/Form1.vb
Imports System.Threading

Public Class Form1

    Public Sub New()
        InitializeComponent()
        AddHandler Application.Idle, AddressOf OnLoaded
        'Update the splashscreen to tell the user we are loading data
        Splashscreen.SplashscreenController.UpdateStatusLabel("Loading data...")
        Thread.Sleep(5000) 'Simulate loading data from an external DB / Webservice
    End Sub

    Protected Overridable Sub OnLoaded(ByVal sender As Object, ByVal args As EventArgs)
        RemoveHandler Application.Idle, AddressOf OnLoaded
        'Close the Splashscreen
        Splashscreen.SplashscreenController.CloseSplash()
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim ds As DataSet1 = New DataSet1
        ds.ReadXml(Application.StartupPath + "\GoogleLocations.DataSet1.xml")
        UrlBindingSource.DataSource = ds
        URLDropDown.SelectedIndex = 0
    End Sub

    Private Sub FetchButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles FetchButton.Click
        toolStripStatusLabel1.Text = "Fetching Google Map..."
        toolStripStatusLabel1.Visible = True
        toolStripProgressBar1.Visible = True
        webBrowser1.Url = New Uri(URLDropDown.SelectedValue.ToString)
    End Sub

    Private Sub webBrowser1_DocumentCompleted(ByVal sender As System.Object, ByVal e As System.Windows.Forms.WebBrowserDocumentCompletedEventArgs) Handles webBrowser1.DocumentCompleted
        toolStripStatusLabel1.Visible = False
        toolStripProgressBar1.Visible = False
    End Sub
End Class

File: releases/Quicksplash-v1.2/Samples/v2.0/CSharp-ProgressBarAndStatusLabel/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Splashscreen;

namespace CSharp_ProgressBarAndStatusLabel
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            Application.Idle += new EventHandler(OnLoaded);
        }

        /// <summary>
        /// Custom "OnLoaded" event.  Fired after form finished loaded & displayed
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="args"></param>
        protected virtual void OnLoaded(object sender, EventArgs args)
        {
            Application.Idle -= new EventHandler(OnLoaded);
            
            //Shut down the splashscreen
            SplashscreenController.CloseSplash();
        }

        #region Sample application logic
        private void Form1_Load(object sender, EventArgs e)
        {
            DataSet1 ds = new DataSet1();
            ds.ReadXml(Application.StartupPath + @"\GoogleLocations.DataSet1.xml");
            UrlBindingSource.DataSource = ds;

            URLDropDown.SelectedIndex = 0;
        }

        private void FetchButton_Click(object sender, EventArgs e)
        {
            toolStripStatusLabel1.Text = "Fetching Google Map...";
            toolStripStatusLabel1.Visible = true;
            toolStripProgressBar1.Visible = true;

            webBrowser1.Url = new Uri(URLDropDown.SelectedValue.ToString());

        }

        private void webBrowser1_DocumentCompleted(object sender, WebBrowserDocumentCompletedEventArgs e)
        {
            toolStripStatusLabel1.Visible = false;
            toolStripProgressBar1.Visible = false;
        }
        #endregion
    }
}